apps/wallet/src/ui/app/components/accounts/UnlockAccountModal.tsx

```diff
@@ -38,15 +41,25 @@ export function UnlockAccountModal({ onClose, onConfirm }: UnlockAccountModalPro
 			password: '',
 		},
 	});
-	const onSubmit = (formValues: FormValues) => {
-		// eslint-disable-next-line no-console
-		console.log('TODO: Do something when the user submits the form successfully', formValues);
-		onConfirm();
+	const backgroundService = useBackgroundClient();
+	const onSubmit = async (formValues: FormValues) => {
+		try {
```

Comment:
@williamrobertson13 @pchrysochoidis this might not be necessary at this point, but I updated this to match the `PasswordInputDialog` component. need to double check the designs but we should move forward with this one